torch/_dynamo/variables/lists.py

```diff
@@ -100,6 +101,67 @@ def call_method(
 
         return super(BaseListVariable, self).call_method(tx, name, args, kwargs)
 
+    @staticmethod
+    def generic_list_compare(left, tx, op, right, **options):
+        from .builder import wrap_fx_proxy
+        from .builtin import BuiltinVariable
+        from .tensor import DynamicShapeVariable
+
+        assert not (
+            left.is_python_constant() and right.is_python_constant()
+        ), "Illegal generic list compare on constant lists"
+
+        # Most list-like variables implement comparison ops the same way,
+        # so they can re-use this helper.
+        # There are quirks though, like how `tuple([2]) == torch.Size([2])`,
+        # but `tuple([2]) != list([2])`
+        if len(left.items) != len(right.items):
+            return ConstantVariable(False, **options)
+        if len(left.items) == 0:
+            return ConstantVariable(True, **options)
+
+        # Generic list comparison works by iterating over left aka self and right the compared-to list.
+        # If we hit here, their lengths are the same and they cannot be expressed as python constants.
+        # So, we iterate over the zipped list items.
+        equal = True
+        comps = []
+        for l_r in zip(left.items, right.items):
+            l = l_r[0]
+            r = l_r[1]
+            comp = BuiltinVariable(op).call_function(tx, [l, r], {})
+            list_type = type(comp)
+            if isinstance(comp, variables.TensorVariable):
+                unimplemented("List Comparison for Tensors is not yet supported")
+            comps.append(comp)
+
+        if len(comps) == 1:
+            return comps[0]
+
+        # Initial postiions
+        prev = comps[0]
+        for i in range(1, len(comps)):
+            # Target for comparison
+            curr = comps[i]
+            options = VariableTracker.propagate([prev, curr])
```

Comment:
I believe the entire rest of this function can just be replaced with:
```suggestion
        return functools.reduce(
            lambda a, b: BuiltinVariable(operator.and_).call_function(tx, [a, b], {}),
            comps
        )
```